### PR TITLE
Viem anvil

### PR DESCRIPTION
**Pull request by agualis**
Setup @viem/anvil to automatically run anvil fork with default parameters.

### CODE REVIEW

File: .github/workflows/checks.yml [hunk 1]
```diff
@@ -43,13 +43,11 @@ jobs:
         uses: ./.github/actions/setup
       - name: Install Foundry
         uses: foundry-rs/foundry-toolchain@v1
-      - name: Run Anvil Node
-        run: anvil --fork-url ${{ secrets.ETHEREUM_RPC_URL }} &
       - name: Test
         run: pnpm test:ci
         env:
-          ETHEREUM_RPC_URL: ${{ secrets.ETHEREUM_RPC_URL }}
```
Comment (johngrantuk) [inline .github/workflows/checks.yml:51]: These RPCs will likely be used for example/debug scripts too so unless there's a strong reason can we keep them without the `VITE_` prefix?

**Comment by brunoguerios** [comment]
Awesome work @agualis ! 👏 
I do have a few questions 😅 
How does this actually work? Does the fork automatically runs before each test? Or does it run only once before all tests?
I ask because while implementing nested joins/exits with relayer queries, I need the fork to not be reset before each test
I know it’s an edge case, but I’d like to get a sense on how to handle this.

File: .github/workflows/checks.yml (hunk 1 above)
Comment (agualis) [reply .github/workflows/checks.yml:51]: Fixed by using `import { loadEnv } from 'vite';` in `vitest.config.ts`
